2.ML_model/README.md

```diff
@@ -0,0 +1,89 @@
+
+We use [sklearn.utils.shuffle](https://scikit-learn.org/stable/modules/generated/sklearn.utils.shuffle.html) to shuffle the training data in a consistent way.
+This is necessary because the data as labeled from MitoCheck tends to have phenotypic classes in groups, which can introduce bias into the model.
+
+We use [sklearn.model_selection.StratifiedKFold](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.StratifiedKFold.html) to create stratified training/test data sets for cross validation.
+The training and test data sets that are created have the same distribution of classes.
+This ensures that each class is proportionally represented across all data sets.
+We use `n_splits=10` to create 10 folds for cross-validation.
+
+### B. Model Training
+
+We use [sklearn.linear_model.LogisticRegression](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html) for our machine learning model.
+We use the following parameters for our Logisic Regression model:
+
+- `penalty='elasticnet'`: Use elasticnet as the penalty for our model.
+Elastic-Net regularization is a combination of L1 and L2 regularization methods.
+The mixing of these two methods is determined by the `l1_ratio` parameter which we optimize later.
+- `solver='saga'`: We use the saga solver as this is the only solver that supports Elastic-Net regularization.
+- `max_iter=100`: Set the maximum number of iterations for solver to converge. 100 iterations allows the solver to maximize performance without completing unnecessary iterations.
+
+We use [sklearn.model_selection.GridSearchCV](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html#sklearn.model_selection.GridSearchCV) to perform an exhaustive search for the following parameters:
+
+- `l1_ratio`: Elastic-Net mixing parameter.
+Used to combine L1 and L2 regularization methods.
+- `C`: Inversely proportional to regularization strength.
+
+We use [sklearn.model_selection.cross_validate](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_validate.html) and [sklearn.model_selection.cross_val_predict](https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.cross_val_predict.html) to train and evaluate a logistic regression model with the best parameters found with GridSearchCV.
+
+### C. Model Evaluation
+
+We compute the average score across estimators after running cross validation.
+
+We  use the predictions from `cross_val_predict()` to create a confusion matrix and precision vs class bar plot for the cross-validated model.
+
+### D. Model Interpretation
+
+Because cross validation produces multiple unique models (estimators), it is necessary to average metrics across the estimators to interpret model performance.
+The coefficient matrices created during each fold of cross validation are averaged to create a single `average_coefs` matrix that we analyze.
```

Comment:
We're averaging cross validation coefficients? Why not use the coefficients from the final model?

Many of the cross validation models will not have good performance. The coefficients could be approximately the final weights, if all combinations of parameters are consistently good. If most of the grid has bad models, then this procedure is going to mislead.

This could also be why we're seeing low sparsity, even with the best model L1_ratio=0.8

Reply:
> After cross-validation a final model is usually created with all of the training data. Is this what you mean by the final model? I agree that it would be better to train a final model with the best params found during GridSearchCV on all the data and only use the coefficients of this final model.

Yes, exactly

Reply:
Done in https://github.com/WayScience/phenotypic_profiling_model/pull/2/commits/087b188514f696ba6a7b593e50c28b77c002ab54.